File: leetcode/705/main.py
class MyHashSet:
	def __init__(self):
		# decide how large array will be
		self.size = 5000
		self.hashTable = [None] * self.size

	def getHashValue(self, key: int) -> int:
		return key % self.size

	def add(self, key: int) -> None:	
		hashVal: int = self.getHashValue(key)
		if self.hashTable[hashVal] is not None:
			for val in self.hashTable[hashVal]:
				if val == key:
					return
			self.hashTable[hashVal].append(key)
		else:
			self.hashTable[hashVal] = [key]

	def contains(self, key: int) -> bool:
		hashVal: int = self.getHashValue(key)
		if self.hashTable[hashVal] is not None: 
			for val in self.hashTable[hashVal]:
				if val == key:
					return True
		return False

	def remove(self, key: int) -> None:		
		hashVal: int = self.getHashValue(key)
		if self.hashTable[hashVal] is not None:
			if len(self.hashTable[hashVal]) == 1 and self.hashTable[hashVal][0] == key:
				self.hashTable[hashVal] = None
			else:
				for idx, val in enumerate(self.hashTable[hashVal]):
					if val == key:
						self.hashTable[hashVal] = self.hashTable[hashVal][:idx] + self.hashTable[hashVal][idx+1:] if idx != len(self.hashTable[hashVal])-1 else self.hashTable[hashVal][:idx]

        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: leetcode/705/main.go
// type MyHashSet struct {
    
// }


// func Constructor() MyHashSet {
    
// }


// func (this *MyHashSet) Add(key int)  {
    
// }


// func (this *MyHashSet) Remove(key int)  {
    
// }


// func (this *MyHashSet) Contains(key int) bool {
    
// }

type MyHashSet struct {
	n int
	table [][]int
}


func Constructor() MyHashSet {
	mhs := new(MyHashSet)
	mhs.n = 50000
	mhs.table = [][]int{}
	for i := 0; i < mhs.n; i++ {
		mhs.table = append(mhs.table, []int{})
	}
	return *mhs
}

func (this *MyHashSet) hashFunc(key int) int {
	return key % this.n
}

func (this *MyHashSet) Add(key int) {
	items := this.table[this.hashFunc(key)]
	for _, item := range items {
		if item == key {
			return 
		}
	}
	this.table[this.hashFunc(key)] = append(this.table[this.hashFunc(key)], key)	
}

func (this *MyHashSet) Remove(key int) {
	items := this.table[this.hashFunc(key)]
	for i, item := range items {
		if item == key {
			if len(items) == 1 {
				this.table[this.hashFunc(key)] = []int{}
			} else {
				this.table[this.hashFunc(key)][i] = items[len(items)-1]
				this.table[this.hashFunc(key)] = items[:len(items)-1]
			}
		}
	}

}

func (this *MyHashSet) Contains(key int) bool {
	items := this.table[this.hashFunc(key)]
	for _, item := range items {
		if item == key {
			return true
		}
	}
	return false
}


/**
 * Your MyHashSet object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Add(key);
 * obj.Remove(key);
 * param_3 := obj.Contains(key);
 */
